matrix/square_matrix.py

```diff
@@ -91,11 +89,97 @@ def LU_decomposition(self):
                 U[i][j] = A[i][j] - sum_upper
                 L[j][i] = (A[j][i] - sum_lower) / U[i][i] if U[i][i] != 0 else 0
 
-        return Matrix(L), Matrix(U)
+        return SquareMatrix(L), SquareMatrix(U)
```

Comment:
Returning SquareMatrix change from last time

Reply:
I was thinking about this further, and I think I'm going to implement a function that based on the inputs will try and figure out which type of matrix to return from the init of the matrix base class, so that we don't need to do this in the future.